hs_rscript_resource/models.py

```diff
@@ -0,0 +1,135 @@
+from django.db import models
+from django.contrib.contenttypes import generic
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor, CoreMetaData, AbstractMetaDataElement
+
+from lxml import etree
+
+
+class RScriptResource(BaseResource):
+    objects = ResourceManager('RScriptResource')
+
+    class Meta:
+        proxy = True
+        verbose_name = 'R Script Resource'
+
+    @classmethod
+    def get_supported_upload_file_types(cls):
+        # one file type is supported
+        return ".r"
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # can have multiple files
+        return True
+
+    @property
+    def metadata(self):
+        md = RScriptMetaData()
+        return self._get_metadata(md)
+
+processor_for(RScriptResource)(resource_processor)
+
+
+class RSMetadata(AbstractMetaDataElement):
+    term = "RSMetadata"
+
+    # program language
+    scriptLanguage = models.CharField(verbose_name='Programming Language', blank=True, max_length=100, default='R',
```

Comment:
Let's go generalized now. I can envision R scripts, Python scripts and
MatLab scripts immediately being used.

On Mon, Jan 4, 2016 at 1:13 PM ironfroggy <notifications@github.com> wrote:

> In hs_rscript_resource/models.py
> <https://github.com/hydroshare/hydroshare/pull/853#discussion_r48774886>:
>
> > +        # can have multiple files
> > +        return True
> > +
> > +    @property
> > +    def metadata(self):
> > +        md = RScriptMetaData()
> > +        return self._get_metadata(md)
> > +
> > +processor_for(RScriptResource)(resource_processor)
> > +
> > +
> > +class RSMetadata(AbstractMetaDataElement):
> > +    term = "RSMetadata"
> > +
> > +    # program language
> > +    scriptLanguage = models.CharField(verbose_name='Programming Language', blank=True, max_length=100, default='R',
>
> I would say don't allow changing the language field but don't name the
> model and classes "RScript" because that's much harder to change in the
> future.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hydroshare/hydroshare/pull/853/files#r48774886>.
>